### PR TITLE
trying to solve issue #858

### PR DESCRIPTION
**Pull request by jaskeerat789**

### CODE REVIEW

File: packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts [hunk 1]
```diff
@@ -85,8 +93,9 @@ export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements G
   public async findBy(filter: Type | AdvancedFilter, page?: GraphbackPage): Promise<Type[]> {
     const { data } = getDatabaseArguments(this.tableMap, filter);
     //tslint:disable-next-line: await-promise
-    const query = this.db.select().from(this.tableName).where(data);
-    const dbResult = await this.usePage(query, page);
+    // const query = this.db.select().from(this.tableName).where(data);
```
Comment (jaskeerat789) [inline packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts]: I am not sure about the implementation of the `WHERE` clause when the data is in the form of an object.

File: packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts [hunk 2]
```diff
@@ -96,7 +105,8 @@ export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements G
   public async batchRead(relationField: string, ids: string[]): Promise<Type[][]> {
     //TODO: Use mapping when relationships are done
     //tslint:disable-next-line: await-promise
-    const dbResult = await this.db.select().from(this.tableName).whereIn(relationField, ids);
+    // const dbResult = await this.db.select().from(this.tableName).whereIn(relationField, ids);
+    const dbResult = await this.db.query(`SELECT * FROM ${this.tableName} WHERE ${relationField} IN (${ids})`) // this is not the correct implimentation for where clause
```
Comment (jaskeerat789) [inline packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts]: same goes for this...  the ids received is an array of ids

**Comment by jaskeerat789** [comment]
I have only made changes in the `KnexDBDataProvider.ts` file as this is there the knex queries were executed.

File: packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts [hunk 3]
```diff
@@ -63,6 +63,21 @@ export class MongoDBDataProvider<Type = any, GraphbackContext = any> implements
     throw new NoDataError(`Cannot update ${this.collectionName}`);
   }
 
+  public async softDelete(data: Type): Promise<Type> {
```
Comment (wtrocki) [inline packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts]: Hi

Can you create separate PR with soft delete change. 
This change is incomplete we will not be able to merge it even if others will work

File: packages/graphback-runtime-knex/src/PgKnexDBDataProvider.ts [hunk 4]
```diff
@@ -11,7 +11,8 @@ import { KnexDBDataProvider } from './KnexDBDataProvider';
  * that works with the rest of the databases.
  */
 //tslint:disable-next-line: no-any
-export class PgKnexDBDataProvider<Type = any, GraphbackContext = any> extends KnexDBDataProvider<Type, GraphbackContext>{
+
+export class PgKnexDBDataProvider<Type = any, GraphbackContext = any> extends PostgreDBDataProvider<Type, GraphbackContext>{
```
Comment (wtrocki) [inline packages/graphback-runtime-knex/src/PgKnexDBDataProvider.ts]: Let's not change any of this code. We should create only `PostgreDBDataProvider` as new file and provide it as an alternative to existing implementations

File: packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts [hunk 5]
```diff
@@ -14,14 +14,14 @@ import { GraphbackDataProvider, GraphbackPage, NoDataError, AdvancedFilter } fro
  * NOTE: For Postgres use dedicated `PgKnexDBDataProvider` that implements more performant creation method.
  */
 //tslint:disable-next-line: no-any
-export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements GraphbackDataProvider<Type, GraphbackContext>{
+export class PostgreDBDataProvider<Type = any, GraphbackContext = any> implements GraphbackDataProvider<Type, GraphbackContext>{
```
Comment (wtrocki) [inline packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts]: Copy this implementation into separate file. 
KnexDataProvider should not be changed

File: packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts (hunk 1 above)
Comment (wtrocki) [reply packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts]: We need to expand parameters with AND

File: packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts (hunk 2 above)
Comment (wtrocki) [reply packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts]: Simply check what queries knex is doing in docker-compose logs

File: packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts (hunk 3 above)
Comment (jaskeerat789) [reply packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts]: oki

File: packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts (hunk 5 above)
Comment (jaskeerat789) [reply packages/graphback-runtime-knex/src/PostgreDBDataProvider.ts]: oki

File: packages/graphback-runtime-knex/src/PgKnexDBDataProvider.ts (hunk 4 above)
Comment (jaskeerat789) [reply packages/graphback-runtime-knex/src/PgKnexDBDataProvider.ts]: ok... I'll create a new PR on this

**Comment by wtrocki** [comment]
I guess this is outdated and new PR will be created?

**Pull request closed without merging**